crates/codegen/language/definition/src/model/manifest.rs

```diff
@@ -144,3 +147,35 @@ pub enum BuiltInLabel {
     LeadingTrivia,
     TrailingTrivia,
 }
+
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+#[derive_spanned_type(Clone, Debug, EnumDiscriminants, ParseInputTokens, WriteOutputTokens)]
+pub enum BuiltIn {
+    BuiltInFunction { item: Rc<BuiltInFunction> },
+    BuiltInType { item: Rc<BuiltInType> },
+    BuiltInVariable { item: Rc<BuiltInField> },
+}
+
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+#[derive_spanned_type(Clone, Debug, ParseInputTokens, WriteOutputTokens)]
+pub struct BuiltInFunction {
+    pub name: String,
+    pub return_type: Option<String>,
+    pub parameters: Vec<String>,
```

Comment:
nit suggestion: ordering:

```suggestion
    pub parameters: Vec<String>,
    pub return_type: Option<String>,
```